### PR TITLE
All controllers now observe Populator, which is Observable

### PR DESCRIPTION
**Pull request by bipvanwinkle**
Populator is now a singleton (allows all controllers to register themselves as observers of the same populator)

### CODE REVIEW

**Comment by bipvanwinkle** [comment]
Don't merge till after Anna merges MapInit

File: java/src/client/communication/ChatController.java [hunk 1]
```diff
@@ -22,4 +27,11 @@ public void sendMessage(String message) {
 
     }
 
+    @Override
+    public void update(Observable o, Object arg) {
+        if (o instanceof Populator && arg instanceof ModelFacade) {
```
Comment (Seiji42) [inline java/src/client/communication/ChatController.java:32]: What would happen here, is you would just go to your Populator/ModelFacade pointer and get the information from there.

File: java/src/client/map/MapController.java [hunk 2]
```diff
@@ -7,25 +7,24 @@
...
     public MapController(IMapView view, IRobView robView) {
 
         super(view);
-        facade = new ModelFacade();
-       
+        Populator.getInstance().addObserver(this);
         setRobView(robView);
 
-        initFromModel();
+        initFromModel(null);
```
Comment (Seiji42) [inline java/src/client/map/MapController.java:27]: Why is null being passed in?

File: java/src/client/model/Populator.java [hunk 3]
```diff
@@ -31,15 +32,24 @@
...
+            instance = new Populator();
+        }
+        return instance;
+    }
 
-    public Populator() {
-        model = new ClientModel();
+    private Populator() {
+        model = Data.getCurentModelInstance();
+        facade = new ModelFacade();
```
Comment (Seiji42) [inline java/src/client/model/Populator.java:47]: Just a thought, initialize Model Facade outside of constructor, and pass in the model (replaces old model)

File: java/src/client/model/Populator.java [hunk 4]
```diff
@@ -55,18 +65,16 @@ public boolean populateModel(ClientModelDTO container) {
         model.setVersion(container.getVersion());
         model.setWinner(container.getWinner());
         
-        Data.initModel(model);
-
-        model.notifyObservers();
+        notifyObservers(facade);
```
Comment (Seiji42) [inline java/src/client/model/Populator.java:68]: Needs function that states that changes have been made

**Comment by bipvanwinkle** [comment]
Its null because the map doesn't need to be initialized until after we have received the model.

**Pull request merged**